jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -192,7 +194,7 @@ static ByteBuffer asByteBuffer(Source source) throws IOException
          */
         static CompletableFuture<byte[]> asByteArrayAsync(Source source, int maxSize)
         {
-            return new ChunkAccumulator().readAll(source, maxSize);
+            return asRetainableByteBuffer(source, null, false, maxSize).thenApply(rbb -> rbb.getByteBuffer().array());
```

Comment:
I've always thought that `as` prefix should be used when returning a different view of the same object.  The `to` prefix should be used when making a new object from the current one.

So this could be thought of as a `to` as it is creates a whole new object.... but that object also mutates the original object (by consuming all its input), so it is kind of a view onto the original object.... if the resulting RBB delays reading the source until it knows how it is going to be used, then it is really is a view onto the source.  So `as` works as well.

I'm enough on the fence not to disrupt things by changing the name at this point.